src/tools/kdb/command.h

```diff
@@ -0,0 +1,177 @@
+	if (tmp != NULL)                                                                                                                   \
+	{                                                                                                                                  \
+		if (elektraStrCmp ("never", keyString (tmp)) == 0)                                                                         \
+		{                                                                                                                          \
+			colorMode = CLI_COLOR_NEVER;                                                                                       \
+		}                                                                                                                          \
+		if (elektraStrCmp ("always", keyString (tmp)) == 0)                                                                        \
+		{                                                                                                                          \
+			colorMode = CLI_COLOR_ALWAYS;                                                                                      \
+		}                                                                                                                          \
+		keyDel (tmp);                                                                                                              \
+	}                                                                                                                                  \
+	char * fmtBuffer = elektraMalloc (1024);                                                                                           \
+	if (fmtBuffer)                                                                                                                     \
+	{                                                                                                                                  \
+	}; /* disable unused variable warning  */                                                                                          \
+	if (!isatty (STDOUT_FILENO))                                                                                                       \
+	{                                                                                                                                  \
+		colorMode = CLI_COLOR_NEVER;                                                                                               \
+	}                                                                                                                                  \
+                                                                                                                                           \
+	int logLevel = verbose + debug;                                                                                                    \
```

Comment:
Is that really what we want? I.e. do want `kdb --debug` and `kdb --verbose` to behave the same?

The man page makes it sound like the options should be completely orthogonal, that would mean we need to separate `bool`s not a single `int LogLevel`.

`enum LOG_LEVEL` implies that `debug` should include more messages than `verbose`. That would be:

```c
// 1 + 1 if debug (verbose doesn't matter)
// 1 + 0 if verbose but not debug
// 0 + 0 if neither
int logLevel = (verbose || debug) + debug;
```

IMO all these variants are acceptable, it's just inconsistent and/or not properly documented right now.

Reply:
By default `verbose` is set to whatever `debug` is, if `-v` should be set, `verbose` will be set to 1.

`-d`:
1. `debug = 1`
2. `verbose = debug`
3. _no `-v` so `verbose` stays_
-> `verbose` + `debug` = 2

`-v`:
1. `debug = 0`
2. `verbose = debug`
3. `-v` so `verbose = 1`
-> `verbose` + `debug` = 1

neither:
1. `debug = 0`
2. `verbose = debug`
3. _no `-v` so `verbose` stays_
-> `verbose` + `debug` = 0

This should work, no?

Reply:
Yes, you're right (technically). I missed that `verbose` is initialized with the value of `debug`. It effectively means that `verbose + debug == (verbose || debug) + debug`.

But only because of two things:
1. `opt/arg = none` means that the `proc:/` key for `--verbose` will always set to `1` or not touched by `elektraGetOpts` at all.
2. Currently only the `proc:/` keys created by `elektraGetOpts` are in `KeySet * options`.

If there was e.g. `user:/sw/elektra/kdb/#0/current/get/verbose` which, would be used as a fallback if `proc:/sw/elektra/kdb/#0/current/get/verbose` doesn't exist. If `user:/sw/elektra/kdb/#0/current/get/verbose` was explicitly set to `0`, then your code would produce `verbose = false`. If `debug = true`, it would mean 
```
verbose + debug = 0 + 1 = 1 = CLI_LOG_VERBOSE
```

So for now it's fine, but `(verbose || debug) + debug` works for the general case too.

Reply:
ohh, I see. You're right